flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -538,19 +557,19 @@ public TaskMetricStore getTaskMetricStore(String taskID) {
     @ThreadSafe
     public static class TaskMetricStore extends ComponentMetricStore {
         private final Map<Integer, SubtaskMetricStore> subtasks;
-        private final Map<String, ComponentMetricStore> jmOperators;
+        private final ComponentMetricStore jmOperator;
 
         private TaskMetricStore() {
-            this(new ConcurrentHashMap<>(), new ConcurrentHashMap<>(), new ConcurrentHashMap<>());
+            this(new ConcurrentHashMap<>(), new ConcurrentHashMap<>(), new ComponentMetricStore());
         }
 
         private TaskMetricStore(
                 Map<String, String> metrics,
                 Map<Integer, SubtaskMetricStore> subtasks,
-                Map<String, ComponentMetricStore> jmOperators) {
+                ComponentMetricStore jmOperators) {
             super(metrics);
             this.subtasks = checkNotNull(subtasks);
-            this.jmOperators = checkNotNull(jmOperators);
+            this.jmOperator = checkNotNull(jmOperators);
```

Comment:
Maybe we could rename `jmOperators` to `jmOperator` too.

Reply:
Good catch!